File: solutions/Pascal's-Triangle-II/Pascal's-Triangle-II.py


class Solution:
    # @return a list of integers
    def getRow(self, rowIndex):
        ans=[[] for i in range(rowIndex+1)]
        for i in range(rowIndex+1):
            for j in range(i,-1,-1):
                if j==0:
                    ans[j]=1
                elif j==i:
                    ans[j]=ans[j-1]
                else:
                    ans[j]=ans[j]+ans[j-1]
        return ans
 

File: solutions/Pascal's-Triangle-II/Pascal's-Triangle-II.js
/**
 * @param {number} rowIndex
 * @return {number[]}
 */
var getRow = function (rowIndex) {
  const ans = [1];
  for (let i = 1; i <= rowIndex; i++) {
    ans[i] = 0;
  }
  for (let i = 1; i <= rowIndex; i++) {
    for (let j = i; j >= 1; j--) {
      ans[j] += ans[j - 1];
    }
  }
  return ans;
};